corehq/apps/hqadmin/views.py

```diff
@@ -259,6 +261,40 @@ def _to_chart_data(data_dict):
         }
 
 
+@require_superuser_or_developer
+def mass_email(request):
+    if request.method == "POST":
+        form = EmailForm(request.POST)
+        if form.is_valid():
+            subject = form.cleaned_data['email_subject']
+            html = form.cleaned_data['email_body_html']
+            text = form.cleaned_data['email_body_text']
+            real_email = form.cleaned_data['real_email']
+
+            if real_email:
+                recipients = [{
+                    'username': h['username'],
+                    'first_name': h['first_name'] or 'CommCare User',
+                } for h in UserES().web_users().run().hits]
```

Comment:
same comment as before, but you can do `.values_list(['username', 'first_name'])` and it will give you a tuple. You would then need to filter out the blank first names in the task, but you are already looping through everything there.

Reply:
Seems like six of one, half dozen of the other, going to leave this alone. This isn't code I expect to live for long.